### ISSUE TITLE
"TypeScript: Restart TS Server" no longer displayed in command palette

### ISSUE THREAD

**Issue by jcready** [microsoft/vscode#210649]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.88.1 (Universal)
Commit: e170252f762678dec6ca2cc69aba1570769a5d39
Date: 2024-04-10T17:42:52.765Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin x64 23.4.0

### Expected Behavior
The "TypeScript: Restart TS Server" option is always available in the command palette when focused on an editor with the language mode set to "TypeScript" or "TypeScript JSX".

### Actual Behavior
The "TypeScript: Restart TS Server" disappears from the command palette sometimes and doesn't return until the window is reloaded.

Steps to Reproduce:

It's hard to know which of these steps or what part of my setup is contributing to this, but I'm working on a remote monorepo which contains many different TS projects that each have their own tsconfig file.

1. Open remote monorepo workspace
2. Open a .ts file.
3. Reload the window. (There should still be the one tab open for the .ts file from step 2)
4. Wait for the "Initializing JS/TS language features" status bar indicator to stop appearing.
5. Open command palette and see that the option to restart the TS server is present.
6. Select/open another .ts file (even one that lives in the same directory as the currently open one).
7. Open command palette and see that the option to restart the TS server is **no longer** present.

This is a duplicate of #206654 which was closed by the author as they apparently aren't experiencing this issue anymore, but I continue to be plagued by it.

Yes, both files are TS files. Yes, the langage mode is set to "TypeScript" for both files. It also happens for the "TypeScript JSX" langage mode.

I can also reproduce this by replacing step 6 with "open a new tab and then switch back to the original .ts tab".

If I "Developer: Reload Window" then all the normal TS commands reappear.
![image](https://github.com/microsoft/vscode/assets/358876/6f783ff1-08d9-49d2-bd5b-4b51604d12cf)

After reloading the window
![image](https://github.com/microsoft/vscode/assets/358876/d9d888bb-d3ea-4d75-9514-f0181f6e89f4)

-----------------

As a complete side note, the issue template said to:
>Instead of creating your report here, use 'Report Issue' from the 'Help' menu in VS Code to pre-fill useful information.

I tried this, but after filling out the whole thing nothing happens when I tried clicking the "Create on Github" button so I am forced to create this issue directly on Github. 🤷

**Comment by BABA983** [comment]
It seems the "TypeScript: Restart TS Server" command only shows when `typescript isManagedFile`.
![ts-language-server/package.json](https://github.com/microsoft/vscode/assets/38986298/7dd471ba-54c0-4d03-82fe-2b042347185a)

**Comment by james-kaguru** [comment]
@Onaxim 
Ive also noted this as well

**Comment by dangerman** [comment]
@Onaxim I tried that scenario and found that:
 * if I focused on the **typescript** editor, the Typescript commands are missing
 * if I focused on the **Markdown** editor, the Typescript commands show up!
 
https://github.com/user-attachments/assets/62c85ade-96cf-4f31-900c-129cff946f93

@BABA983 I searched for the `typescript.isManagedFile` using the `Check Context Keys` dev command on each of the editors:
 * in the **typescript** editor it was `false`
 
<img width="1156" alt="Screenshot 2024-07-23 at 00 11 28" src="https://github.com/user-attachments/assets/f1e69438-06f7-4d5e-abf2-f08ef316103d">

 * in the **Markdown** editor it was `true`
 
<img width="1148" alt="Screenshot 2024-07-23 at 00 07 08" src="https://github.com/user-attachments/assets/45b5ba93-9e76-4d1c-9eae-ecfb42af5428">

**Comment by dangerman** [comment]
I suspect this might be a regression introduced by https://github.com/microsoft/vscode/pull/201847 which is itself a fix for https://github.com/microsoft/vscode/issues/201734 (missing 'Restart TS Server' command when viewing diff editor) which is currently still open.

When debugging locally I found that `ManagedFileContextManager.onDidChangeActiveTextEditor()` seemed to be getting the **wrong editor**, which would explain the TS commands only being visible when I had the MD editor open.
https://github.com/microsoft/vscode/blob/22e3447b4b410dcf8426bf70743c5dd4030d6122/extensions/typescript-language-features/src/ui/managedFileContext.ts#L28-L30

So I checked the `ActiveJsTsEditorTracker` and found that in the `getEditorCandidatesForActiveTab()` method it tries to get the active editor using tabs via `vscode.window.tabGroups.activeTabGroup.activeTab` but at runtime this seems to be returning the wrong value...

<img width="1312" alt="active-tab" src="https://github.com/user-attachments/assets/3f03989a-e74e-4bff-802a-740dc3268962">

**Comment by dangerman** [comment]
I think I might have found a potential fix for this - well, @Onaxim's scenario at least.

The `ActiveJsTsEditorTracker` calls its `update()` method on the following events:
 * onDidChangeActiveTextEditor
 * onDidChangeVisibleTextEditors
 And I previously observed that `vscode.window.tabGroups.activeTabGroup` was incorrect.
 
 However, maybe at that point, the tab group state wasn't updated yet. So I tried subscribing to the `tabGroups.onDidChangeTabGroups` event, and the `activeTabGroup` does resolve correctly in that case. 

https://github.com/user-attachments/assets/4b4bebc2-7a0d-443e-8a75-28b7a0aad8b5

I've raised a draft PR for now: https://github.com/microsoft/vscode/pull/223433
